src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/OperatorWrapCheck.java

```diff
@@ -302,28 +304,150 @@ public void setOption(String optionStr) {
 
     @Override
     public void visitToken(DetailAST ast) {
-        final DetailAST parent = ast.getParent();
-        // we do not want to check colon for cases and defaults
-        if (parent.getType() != TokenTypes.LITERAL_DEFAULT
-                && parent.getType() != TokenTypes.LITERAL_CASE) {
-            final String text = ast.getText();
-            final int colNo = ast.getColumnNo();
-            final int lineNo = ast.getLineNo();
-            final String currentLine = getLine(lineNo - 1);
-
-            // Check if rest of line is whitespace, and not just the operator
-            // by itself. This last bit is to handle the operator on a line by
-            // itself.
+        if (isOperatorNode(ast)) {
             if (option == WrapOption.NL
-                    && !text.equals(currentLine.trim())
-                    && CommonUtil.isBlank(currentLine.substring(colNo + text.length()))) {
-                log(ast, MSG_LINE_NEW, text);
+                    && TokenUtil.areOnSameLine(ast, getLeftNode(ast))
```

Comment:
is it more readable form:
```
if (isOperatorNode(ast)) {
    if (option == WrapOption.NL) {
        if (TokenUtil.areOnSameLine(ast, getLeftNode(ast))
              && !TokenUtil.areOnSameLine(ast, getRightNode(ast))) {
           log(ast, MSG_LINE_NEW, ast.getText());
        }
    }
    else if (option == WrapOption.EOL) {
      if (!TokenUtil.areOnSameLine(ast, getLeftNode(ast))) {
        log(ast, MSG_LINE_PREVIOUS, ast.getText());
      }
    }
}

```

user who debug certain option will not need to real long expressions at all.

Reply:
This lead to PMD violation https://pmd.github.io/latest/pmd_rules_vm_design.html#collapsibleifstatements
Maybe it is possible to fool this rule with a switch.

Reply:
Upd: fixed with two additional methods

Reply:
we can suppress Pmd on this case, it is normal that tools have false positives.
we should focus on what is good for reading.